soh/src/overlays/actors/ovl_En_Partner/z_en_partner.c

```diff
@@ -850,14 +862,28 @@ void EnPartner_Draw(Actor* thisx, PlayState* play) {
 
     gSPSegment(POLY_XLU_DISP++, 0x08, dListHead);
     gDPPipeSync(dListHead++);
-    gDPSetPrimColor(dListHead++, 0, 0x01, (u8)this->innerColor.r, (u8)this->innerColor.g, (u8)this->innerColor.b,
-                    (u8)(this->innerColor.a * alphaScale));
+    if (CVarGetInteger("gCosmetics.Ivan_IdlePrimary.Changed", 0)) {
+        Color_RGB8 ivanColor1 = CVarGetColor24("gCosmetics.Ivan_IdlePrimary.Value", (Color_RGB8){ 255, 255, 255 });
+        gDPSetPrimColor(dListHead++, 0, 0x01, (u8)ivanColor1.r, (u8)ivanColor1.g, (u8)ivanColor1.b,
+                        (u8)(this->innerColor.a * alphaScale));
```

Comment:
Since the body and particle colors are the same (both default green or the cosmetic color), instead of doing this lookup twice in each draw func, you can instead do it once in the `EnPartner_Update` func, and have it set `this->innerColor` and `this->outerColor` to the right color, then the two draw funcs can be reverted to use those values again.

Reply:
> Since the body and particle colors are the same (both default green or the cosmetic color), instead of doing this lookup twice in each draw func, you can instead do it once in the `EnPartner_Update` func, and have it set `this->innerColor` and `this->outerColor` to the right color, then the two draw funcs can be reverted to use those values again.

![image](https://github.com/HarbourMasters/Shipwright/assets/105878730/05e10f40-4897-4beb-af56-f7a81bcc7aae)


TY for your help and feedback :D

1. How can I get the colors from Cosmetics Editor with a function similar to `CVarGetColor24()` that returns a `Color_RGBAf` value instead of returning a `Color_RGB8` value?

-or-

2. Should I instead modify the definitions of `this->innerColor` and `this->outerColor` to be of type `Color_RGB8` so I can assign the Cosmetic Editor colors via the `CVarGetColor24()` function?

Thank you in advance :)

Reply:
You can assign the individual color properties one by one, which will handle the casting from int to float.

```c
Color_RGB8 idlePrim = CVarGetColor24(...);
this->innerColor.r = idlePrim.r;
this->innerColor.g = idlePrim.g;
...
```

Or if you are daring, you can change the `Color_RGBAf` in the EnPartner struct to be `Color_RGBA8`. Either way is fine.